### DISCUSSION TITLE
How to refresh a select field in the main form after the value was changed in the relationship manager

### DISCUSSION THREAD

**Discussion by wq9578** [filamentphp/filament#10691] [Help]
How can I refresh a select field in the main form after the value was changed in the relationship manager?
The simple dispatch/listener approach doesn't work.

Setting:
When editing a **person**, one can add an **address** in the relationship manager shown below.
If this is the first address created, then `$person->default_address_id` is updated accordingly to this first and only address record.
But the select field (for the default address) then still displays "Select an option" (when clicking the select field, however, the new address is shown below that, which means that a refresh took place).
The desired behaviour would be a select field (for the default address) displaying the new address, to which `$person->default_address_id` is pointing to. (This desired display can be enforced with a manual reload in the browser.) 
Even aggressively calling refreshFormData() or fillForm() doesn't help (usually an empty public function refresh() is sufficient).

Code:
```
namespace App\Filament\Test\Resources\PersonResource\RelationManagers;

...

class AddressesRelationManager extends RelationManager

...

    public function table(Table $table): Table

...

            ->headerActions([
                Tables\Actions\CreateAction::make()
                    ->after(function (Address $address, Component $livewire) {
                        $person = $address->person;

                        if ($person->addresses->count() == 1) {
                            $person->default_address_id = $person->addresses->first()->id;

                            $person->save();

                            $livewire->dispatch('refreshPerson');
                        }
                    }),
            ])
...
```

```
namespace App\Filament\Test\Resources\PersonResource\Pages;

...

class EditPerson extends EditRecord
{
    protected static string $resource = PersonResource::class;

    #[On('refreshPerson')]
    public function refresh(): void
    {
        $this->refreshFormData(['default_address_id']);

        $this->fillForm();
    }

...
```

**Comment by akulmambetov** [comment]
The same problem

**Comment by peters-cmc** [comment]
In v2 I recall using `->loadStateFromRelationships()` on the component to be updated. Not sure if this would help here (worked for me on Repeaters in v2).

**Comment by imknight** [comment]
I use the same refreshFormData and FillForm in the afterSave method and it work.

**Comment by myregistration** [comment]
I had some wonky stuff happen with `DateTimePicker` fields until I used  `$this->fillForm() ` after the `->refreshFormData(['datefield_1', 'datefield_2'])`.  It appears to be working correctly now so hopefully that's all it was.
